File: Molinos.Scato.Dominio/Entidades/NominacionDatoTecnico.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Molinos.Scato.Dominio.Entidades
{
    public class NominacionDatoTecnico : IIdentificable
    {
        [Key]
        public virtual int Id { get; set; }
        public virtual MaterialPuerto MaterialPuerto { get; set; }
        public virtual decimal CantidadTotal { get; set; }
        public virtual int Tolerancia { get; set; }
        public virtual string Observaciones { get; set; }
        public virtual VaporInformacion VaporInformacion { get; set; }
        public virtual DateTime? ETARecalada { get; set; }
        public virtual DateTime? ObligacionDeCarga { get; set; }
        public virtual MuelleDeCarga MuelleDeCarga { get; set; }
        public virtual TasaDeCarga TasaDeCarga { get; set; }
        public virtual decimal? TasaDeCargaValor { get; set; }
        public virtual decimal DEM { get; set; }
        public virtual decimal DES { get; set; }
        public virtual TipoDeContrato TipoDeContrato { get; set; }
        public virtual ATAPuerto ATAPuerto { get; set; }
        public virtual AgenciaMaritimaPuerto AgenciaMaritimaPuerto { get; set; }
        public virtual Surveyor Surveyor { get; set; }
        public virtual string ObservacionesSurveyor { get; set; }
        public virtual ICollection<NominacionDatoTecnicoCalidad> NominacionDatoTecnicoCalidad { get; set; }
        public virtual ICollection<NominacionDatoTecnicoCoordinadorPuerto> NominacionDatoTecnicoCoordinadorPuerto { get; set; }
        public virtual ICollection<NominacionDatoTecnicoExportador> NominacionDatoTecnicoExportador { get; set; }
        public virtual ICollection<NominacionDatoTecnicoDestino> NominacionDatoTecnicoDestino { get; set; }
        public virtual string OtroMuelleNombre { get; set; }
    }
}

File: Molinos.Scato.WebPuerto/src/app/shared/modelos/programa-embarque/nominacion-dato-tecnico.ts
import { MaterialPuerto } from "../material-puerto";
import { AgenciaMaritimaPuerto } from "../agencia-maritima-puerto";
import { Vapor } from "../vapor";
import { ATAPuerto } from "../ata-puerto";
import { TasaDeCarga } from "./tasa-de-carga";
import { MuelleDeCarga } from "./muelle-de-carga";
import { Surveyor } from "./surveyor";
import { TipoDeContrato } from "./tipo-de-contrato";
import { NominacionDatoTecnicoCalidad } from "./nominacion-dato-tecnico-calidad";
import { VaporInformacion } from "@ScatoModels/Buques/VaporInformacion";
import { NominacionDatoTecnicoDestino } from "./nominacion-dato-tecnico-destino";
import { NominacionDatoTecnicoExportador } from "./nominacion-dato-tecnico-exportador";
import { NominacionDatoTecnicoCoordinador } from "./nominacion-dato-tecnico-coordinador";


export class NominacionDatoTecnico {
    id: number;
    materialPuerto: MaterialPuerto;
    cantidadTotal: number;
    cantidadExacta: number;
    cantidadConTolerancia: number;
    tolerancia: number;
    cantidadTotalMaxima: number;
    observaciones: string;
    vaporInformacion: VaporInformacion;
    etaRecalada: Date;
    obligacionDeCarga: Date;
    muelleDeCarga: MuelleDeCarga;
    tasaDeCarga: TasaDeCarga;
    tasaDeCargaValor: number;
    dem: number;
    des: number;
    tipoDeContrato: TipoDeContrato;
    ataPuerto: ATAPuerto;
    agenciaMaritimaPuerto: AgenciaMaritimaPuerto;
    surveyor: Surveyor;
    observacionesSurveyor: string;
    nominacionDatoTecnicoCalidad: NominacionDatoTecnicoCalidad[];
    nominacionDatoTecnicoDestino: NominacionDatoTecnicoDestino[];
    nominacionDatoTecnicoExportador: NominacionDatoTecnicoExportador[];
    nominacionDatoTecnicoCoordinadorPuerto: NominacionDatoTecnicoCoordinador[];
    otroMuelleNombre: string;

    constructor(id, materialPuerto, cantidadTotal,
        tolerancia, observaciones, vaporInformacion,
        etaRecalada, obligacionDeCarga, muelleDeCarga,
        tasaDeCarga, tasaDeCargaValor, dem,
        des, tipoDeContrato, ataPuerto,
        agenciaMaritimaPuerto, surveyor, observacionesSurveyor, nominacionDatoTecnicoCalidad,
        otroMuelleNombre: string = '', cantidadExacta, cantidadConTolerancia, cantidadTotalMaxima
    ) {
        this.id = id;
        this.materialPuerto = materialPuerto;
        this.cantidadTotal = cantidadTotal;
        this.tolerancia = tolerancia;
        this.observaciones = observaciones;
        this.vaporInformacion = vaporInformacion;
        this.etaRecalada = etaRecalada;
        this.obligacionDeCarga = obligacionDeCarga;
        this.muelleDeCarga = muelleDeCarga;
        this.tasaDeCarga = tasaDeCarga;
        this.tasaDeCargaValor = tasaDeCargaValor;
        this.dem = dem;
        this.des = des;
        this.tipoDeContrato = tipoDeContrato;
        this.ataPuerto = ataPuerto;
        this.agenciaMaritimaPuerto = agenciaMaritimaPuerto;
        this.surveyor = surveyor;
        this.observacionesSurveyor = observacionesSurveyor;
        this.nominacionDatoTecnicoCalidad = nominacionDatoTecnicoCalidad;
        this.otroMuelleNombre = otroMuelleNombre;
        this.cantidadConTolerancia = cantidadConTolerancia;
        this.cantidadExacta = cantidadExacta;
        this.cantidadTotalMaxima = cantidadTotalMaxima;
    }
}
